src/codegen/compile.ml

```diff
@@ -6053,7 +6053,7 @@ module Internals = struct
     | Some (VarEnv.Const(_, Const.Fun mk_fi)) ->
        compile_unboxed_zero ^^ (* A dummy closure *)
        G.i (Call (nr (mk_fi ())))
-    | _ -> assert false
+    | _ -> failwith var
```

Comment:
If you leave in `failwith`, maybe with a better error message?

Reply:
Nope, this is a change I did for debugging only. I'll revert!

Reply:
```suggestion
    | _ -> assert false
```